src/app.ts

```diff
@@ -13,14 +13,40 @@ interface TemperatureSummary {
   average: number
 }
 
+interface TemperatureArray {
+  [key: string]: number[]
+}
+
+const summary: TemperatureArray = {}
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach((element: TemperatureReading) => {
+    const nameElement = `${element.city + element.time.toDateString()}`
+
+    summary[nameElement] ??= []
+
+    summary[nameElement].push(element.temperature)
+  })
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const nameElement = `${city + date.toDateString()}`
+
+  const onlyTemperatures = summary[nameElement]
+
+  if (!summary[nameElement]) return null
+
+  const count = onlyTemperatures.length
+
+  return {
+    first: onlyTemperatures[0],
+    last: onlyTemperatures[count - 1],
+    high: Math.max(...onlyTemperatures),
+    low: Math.min(...onlyTemperatures),
```

Comment:
Internally, `Math.max` and `Math.max` will apply a `sort` in order to get the maximum and minimum values, so you could have leveraged using by yourself a `sort` and then get the first and the last value `result[0]` and `result[result.length - 1]`